src/utils/safeHref.ts

```diff
@@ -0,0 +1,102 @@
+export interface URLInfo {
+    url: string | undefined
+    sameOrigin: boolean
+}
+
+export interface SafeHrefConfigObj {
+    enabled?: boolean
+    origin?: string
+}
+
+const PROTOCOL_REGEX = /^[a-z]+:/
+const ORIGIN_REGEX = /^(?:[a-z]+:?:)?(?:\/\/)?([^\/\?]+)/
+let useSafeHref = false
+let globalOrigin = typeof window !== null ? window.location.origin : false
```

Comment:
I think this has to check for `undefined`: https://github.com/segmentio/evergreen/blob/26173dff47ae58af16b15b80c0083c8a6ce66840/src/ssr/src/autoHydrate.js#L19

Reply:
Also `typeof` always returns a string so this would never have matched.

Reply:
thats a good point, typo on my part

Reply:
fixed 😄